scripts/elior.coffee

```diff
@@ -1,15 +1,32 @@
 'use strict'
 
-fixCase = (str) -> str.charAt(0).toUpperCase() + str.slice(1).toLowerCase()
-
 module.exports = (robot) ->
-  robot.respond /menu/, (chat) ->
+  getMenu = (chat, day) ->
+    formatMenu = (dayMenu) ->
+      formatDay = () -> dayMenu.date.replace 'T00:00:00Z', ''
+
+      formatPlats = () ->
+        fixCase = (str) -> str.charAt(0).toUpperCase() + str.slice(1).toLowerCase()
+        (dayMenu.famillePlats[0].plats.map (p) -> fixCase(p.libelle)).join ', '
+
+      "Menu du #{formatDay dayMenu} : #{formatPlats dayMenu}"
+
     robot.http("https://timechef.elior.com/api/restaurant/1271/menus")
       .get() (err, res, body) ->
         if err
           console.log "error: " + err
         else
           data = JSON.parse body
-          plats = (data[0].famillePlats[0].plats.map (p) -> fixCase(p.libelle)).join ', '
-          day = data[0].date.replace 'T00:00:00Z', ''
-          chat.reply "Menu du #{day} : #{plats}"
+          if day < 0
+            chat.reply (data.map (dayMenu) -> formatMenu dayMenu).join '\n'
+          else if day < 5
+            chat.reply formatMenu data[day]
+
+  robot.respond /menu$/, (chat) ->
+    getMenu chat, 0
```

Comment:
I would prefer having getMenu keep to a single role: get the menu and return it as a string. It's up to the caller to decide what to do with this string then:

 robot.respond /menu$/, (chat) ->
   chat.reply (getMenu day)